src/mono/wasm/runtime/js_support.js

```diff
@@ -0,0 +1,25 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// THIS FILE IS COPIED DIRECTLY INTO THE DOTNET.JS FILE WITHOUT BEING RUN OR COMPILED/OPTIMZIED
+// IT I MEANT AS A SET OF JS TOOLS TO SUPPORT THE SDK
+
+var JSSupportLib = {
+    // Loads the config file located in the root of the project
+    load_config: function (callback) {
```

Comment:
we should probably check that `callback` is defined

Reply:
What should the function do if the callback is not defined? Should it throw an exception or does wasm have some other way of handling such cases?

Reply:
I think it should invoke the `callback` only if it is defined, and ignore it otherwise. I think you already have a case where no `callback` is being passed - https://github.com/dotnet/runtime/pull/53248/files#diff-d9682df4e3d8adce23feb8821c759d307113d4878a84f185e909f17ff8d872fbR218 .

Looking at that now, I wonder can we have a situation where `onRuntimeInitialized` gets called but we don't have the config loaded yet?

Reply:
Yes we can. In fact that is what happens each time. I call to load the config after the runtime has initialized by using 
```
onRuntimeInitialized: function () {
      JSSupportLib.load_config(this.onConfigLoaded);
},
```
This is what I meant with the open question. It seems a bit cumbersome to require the user to load the config manually like this.

Reply:
Resolved the open question in next commit